mmv1/third_party/terraform/provider/provider_access_token_test.go

```diff
@@ -0,0 +1,306 @@
+package provider_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+// TestAccSdkProvider_access_token is a series of acc tests asserting how the SDK provider handles access_token arguments
+// It is SDK specific because the HCL used provisions SDK-implemented resources
+// It is a counterpart to TestAccFwProvider_access_token
+func TestAccSdkProvider_access_token(t *testing.T) {
+	testCases := map[string]func(t *testing.T){
+		"config takes precedence over environment variables":                                                        testAccSdkProvider_access_token_configPrecedenceOverEnvironmentVariables,
+		"when access_token is unset in the config, environment variables are used in a given order":                 testAccSdkProvider_access_token_precedenceOrderEnvironmentVariables, // GOOGLE_OAUTH_ACCESS_TOKEN
+		"when access_token is set to an empty string in the config the value isn't ignored and results in an error": testAccSdkProvider_access_token_emptyStringValidation,
+		"access_token conflicts with credentials":                                                                   testAccSdkProvider_access_token_conflictsWithCredentials,
+		"access_token can be used to authenticate the provider":                                                     testAccSdkProvider_access_token_authInUse,
```

Comment:
```suggestion
		// Configuring the provider using inputs
		"config takes precedence over environment variables":                                                        testAccSdkProvider_access_token_configPrecedenceOverEnvironmentVariables,
		"when access_token is unset in the config, environment variables are used in a given order":                 testAccSdkProvider_access_token_precedenceOrderEnvironmentVariables, // GOOGLE_OAUTH_ACCESS_TOKEN
		// Schema-level validation		
		"when access_token is set to an empty string in the config the value isn't ignored and results in an error": testAccSdkProvider_access_token_emptyStringValidation,
		"access_token conflicts with credentials":                                                                   testAccSdkProvider_access_token_conflictsWithCredentials,
		// Usage
		"access_token can be used to authenticate the provider":                                                     testAccSdkProvider_access_token_authInUse,
```
		"config takes precedence over environment variables":                                                        testAccSdkProvider_access_token_configPrecedenceOverEnvironmentVariables,
		"when access_token is unset in the config, environment variables are used in a given order":                 testAccSdkProvider_access_token_precedenceOrderEnvironmentVariables, // GOOGLE_OAUTH_ACCESS_TOKEN
		"when access_token is set to an empty string in the config the value isn't ignored and results in an error": testAccSdkProvider_access_token_emptyStringValidation,
		"access_token conflicts with credentials":                                                                   testAccSdkProvider_access_token_conflictsWithCredentials,
		"access_token can be used to authenticate the provider":                                                     testAccSdkProvider_access_token_authInUse,
```

Reply:
d180551